packages/logger/src/logger.ts

```diff
@@ -0,0 +1,49 @@
+import { fromReaderIOK as toReaderTaskEitherK } from "fp-ts/ReaderTaskEither";
+import { pipe } from "fp-ts/function";
+
+export type LogRecord = {
+  timestamp: Date;
+  level: "debug" | "info" | "warn" | "error" | "fatal";
+  message: string;
+  [context: string]: unknown;
+};
+
+export type Logger = {
+  log: (s: string, l: LogRecord["level"]) => () => void;
+  format: (r: LogRecord) => string;
+  context?: Record<string, unknown>;
+};
+
+export const log =
+  (level: LogRecord["level"]) =>
+  (message: LogRecord["message"], context: Record<string, unknown> = {}) =>
+  (r: { logger: Logger }) =>
```

Comment:
I would add a comment on this part. What is `r`?

Reply:
`r` is the dependencies object, `r` as `Reader`

Reply:
I think it's necessary to always log the error like debugging the request.